handlers/hormuud/handler.go

```diff
@@ -155,16 +151,18 @@ func (h *handler) FetchToken(ctx context.Context, channel courier.Channel, msg c
 	req.Header.Set("Accept", "application/json")
 
 	resp, respBody, err := h.RequestHTTP(req, clog)
-	if err != nil || resp.StatusCode/100 != 2 {
-		return "", errors.Wrapf(err, "error making token request")
+	if err != nil {
+		return "", courier.ErrConnectionFailed
+	} else if resp.StatusCode/100 != 2 {
+		return "", courier.ErrResponseStatus
```

Comment:
this check isn't quite right in so many handlers... maybe we should tweak `RequestHTTP` to return`ErrConnectionFailed` or  `ErrResponseStatus`... is there ever a case where we need to handle a non-200 response differently?

Reply:
We can check on that

Reply:
I guess the issue is that some handlers want to parse a 400 response to extract an error code

Reply:
Well either way I think you need to do an audit of `RequestHTTP` calls...